### PR TITLE
Fix/make docker

### PR DESCRIPTION
**Pull request by oelbert**
**Description**
Adds a dockerfile for testing and developing the physics code, updates the makefile to work with the current data and docker setup.

- `Dockerfile`: Added, contains a recipe for an image that can run the code
- `Makefile`: Replaced with version that pulls data and builds images correctly without ai2/vcm dependencies
- `examples/notebook/test_physics.ipynb`: Added an example notebook that runs a stencil of pySHiELD code with NDSL infrastructure
- `setup.py`: adds pyFV3 as it is a dependency

**How Has This Been Tested?**
Savepoint tests have been run via docker, notebook executes pyshield and ndsl code

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] New check tests, if applicable, are included
- [ ] Targeted model, if this change was triggered by a model need/shortcoming

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "f90nml>=1.1.0",
     "numpy",
     "xarray",
+    "pyFV3 @ git+https://github.com/NOAA-GFDL/PyFV3.git@main",
```
Comment (FlorianDeconinck) [inline setup.py]: Should `pyFV3`  be part of the base reqs ? You actually have a `pyFV3_requirements` down below, which can be installed with `pySHiELD[pyFV3]`

**Review by FlorianDeconinck — CHANGES_REQUESTED** [review]
`pyFV3` install probably needs to change, rest is good.

File: setup.py (hunk 1 above)
Comment (oelbert) [reply setup.py]: Fair point. I was concerned by having imports that aren't instaled by default, but I guess if pyFV3 is only imported in the "update" code it's fine

Comment (FlorianDeconinck) [reply setup.py]: That update code should not exist here though, or at least not with using the `pyFV3` repository. Right now `pySHiELD` is only plug-able with a `pyFV3` dycore state, which arguably is not an issue considering that their isn't a ton of dycore floating around, but architecturally we should abstract that layer away

**Review by FlorianDeconinck — APPROVED** [review]
I logged the notebook under `ci` task.

LGTM

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,64 @@
+FROM python:3.8
```
Comment (mlee03) [inline Dockerfile:1]: Just to keep the number of pace containers in check, would it work with `FROM ghcr.io/noaa-gfdl/pace:3.8`

That container has 

gcc 13.2.0, 
netcdf-c 4.9.2 (hdf5 1.14.2)
netcdf-fortran 4.6.1
mpich 4.1.2
git 2.42.0
miniconda 22.11.1

installed.  The container can be rebuilt with openmpi in place of mpich; and with different version of packages (as long as they are availableon spack)

Comment (mlee03) [reply Dockerfile:1]: Also should add it also has GFDL's FRE-NCTools installed so grids can be generated on the fly

Comment (fmalatino) [reply Dockerfile:1]: I think the default for mpi should be openmpi as that is what we have been using on the development box and gaea

**Comment by bensonr** [comment]
As many of the larger computer systems we work with are currently HPE/Cray systems with a vendor implementation of MPICH, it wouldn't be a bad idea to test both OpenMPI and MPICH.  Perhaps we should discuss at the next NASA-NOAA tag-up.

File: Dockerfile (hunk 2 above)
Comment (oelbert) [reply Dockerfile:1]: I'm not opposed to that, as long as it's broadly available. Can you point me to the dockerfile for the image and I'll try building using that as a base

Comment (mlee03) [reply Dockerfile:1]: The dockerfile is [here](https://github.com/mlee03/pace/blob/pace-ci-testing/Dockerfile).  The already built image can be pulled with `docker pull ghcr.io/noaa-gfdl/pace:3.8`.  I'll push the image with openmpi installed next week to the registry.

Comment (bensonr) [reply Dockerfile:1]: As many of the larger computer systems we work with are currently HPE/Cray systems with a vendor implementation of MPICH, it wouldn't be a bad idea to test both OpenMPI and MPICH. Perhaps we should discuss at the next NASA-NOAA tag-up.

**Review by fmalatino — APPROVED** [review]
Approved pending inclusion and testing with mpich

File: Dockerfile (hunk 2 above)
Comment (oelbert) [reply Dockerfile:1]: I'm having a bit of trouble building the docker image from ghcr.io/noaa-gfdl/pace:3.8 on my mac. I'd also like to keep this similar. to the pyFV3 container as well. Would it be alright to keep this container as-is?

Comment (mlee03) [reply Dockerfile:1]: Is it the personal token issue?  But totally ok with going with your Dockerfile  (Btw, I discovered mpich in the Github container was compiled incorrectly, working on fixing that as well as creating an openmpi one :grin: )

**Review by mlee03 — APPROVED** [review]

**Comment by oelbert** [comment]
> Approved pending inclusion and testing with mpich

Should I be trying to use mpich in this dockerfile? I thought we were testing with mpich in the pace dockerfile?

**Review by fmalatino — APPROVED** [review]

**Review by fmalatino — APPROVED** [review]
Approved pending inclusion of openmpi and mpich

**Pull request merged**